app.js

```diff
@@ -1,112 +1,110 @@
-};
 
-function showImages() {
-  var imageOneBox = document.getElementById('imageOne');
-  var showImageOne = document.createElement('img');
-  showImageOne.src = productPath[randomImage()];
-  imageOneBox.appendChild(showImageOne);
+var productImageNames = ['bag', 'banana', 'bathroom', 'boots', 'breakfast', 'bubblegum', 'chair', 'cthulhu', 'dog-duck', 'dragon', 'pen', 'pet-sweep', 'scissors', 'shark', 'sweep', 'tauntaun', 'unicorn', 'usb', 'water-can', 'wine-glass'];
+var productsArray = [];
 
+function Products(name) {
+  this.name = name;
+  this.path = './assets/' + name + '.jpg';
+  this.votes = 0;
+  productsArray.push(this)
 
-  var imageTwoBox = document.getElementById('imageTwo');
-  var showImageTwo = document.createElement('img');
-  showImageTwo.src = productPath[randomImage()];
-  imageTwoBox.appendChild(showImageTwo);
+}
+// builds products with iffe
+(function() {
+  for(var i = 0; i < productImageNames.length; i++) {
+    new Products(productImageNames[i]);
+  }
+})();
 
+var tracker = {
+  imageOneEl: document.getElementById('imageOne'),
+  imageTwoEl: document.getElementById('imageTwo'),
+  imageThreeEl: document.getElementById('imageThree'),
+  showResultsEl:document.getElementById('show-results'),
+  resultsEl: document.getElementById('results'),
+  imageContainerEl: document.getElementById('image-container'),
+  imageObjOne: null,
+  imageObjTwo: null,
+  imageObjThree: null,
+  clicks: 1,
 
-  var imageThreeBox = document.getElementById('imageThree');
-  var showImageThree = document.createElement('img');
-  showImageThree.src = productPath[randomImage()];
-  imageThreeBox.appendChild(showImageThree);
-}
 
-var imageContainer = document.getElementById('image-container');
-imageContainer.addEventListener('click', function(){
-  showImages()
-// refreshes pictures
-var refreshImages = document.getElementsByTagName('img');
-for (var i = 0; i < refreshImages.length-1; i++) {
-  refreshImages[0].parentNode.removeChild(refreshImages[0]);
-}
-});
-
-// function tracker() {
-//   var imageContainer = document.getElementById('image-container');
-//   imageContainer.addEventListener('click', function(){
-//     showImages()
-//   });
-//
-//   var refresh = document.getElementById('image-container');
-//   while(productPath.length > 0) {
-//     productPath[0].parentNode.removeChild(productPath[0]);
-//   }
+  getRandomIndex: function() {
+    return Math.floor(Math.random() * productsArray.length);
+  },
+
+  // function that displays 3 random non repeating images
+  displayImages: function() {
+    this.imageObjOne = productsArray[this.getRandomIndex()];
+    this.imageObjTwo = productsArray[this.getRandomIndex()];
+    this.imageObjThree = productsArray[this.getRandomIndex()];
+
+    if(this.imageObjOne === this.imageObjTwo || this.imageObjOne === this.imageObjThree || this.imageObjTwo === this.imageObjThree){
+      this.displayImages();
+    }
+    this.imageOneEl.src = this.imageObjOne.path;
```

Comment:
When you have a repeated task like this, how do you normally accomplish it without hardcoding it every time?